lib/response.js

```diff
@@ -1052,6 +1052,32 @@ function sendfile(res, file, options, callback) {
 }
 
 /**
+ * Run every tracers set at the application level for given events
+ * and arguments. To set a tracer on the application use the `instrument`
+ * method.
+ *
+ * @param {function} event The event to trace.
+ * @return {ServerResponse} for chaining
+ * @public
+ */
+
+res.trace = function(event) {
+  var args = [];
+  switch (arguments.length) {
+    case 0:
+      throw new Error('No event defined!');
+    case 1: break;
+    case 2:
+      args = [arguments[1]];
+      break;
+    default:
+      args = [].slice.call(arguments, 1);
+  }
+  this.app._callTracers(this, event, args);
```

Comment:
Ah yes, @No9 that reminds me about something talked about in a recent Express TC meeting: docs. Normally on all repos, adding documentation is a requirement to merge a PR, but with this repo, where the docs are not in the same repo, it's not easy to do that. A suggest was that we should actually make having a PR in https://github.com/expressjs/expressjs.com a blocker to get something merged here, so we know there are actually docs ready to roll when we release the feature, rather than just a promise to maybe make them in the future :) @frankrousseau would need to do this most likely, as otherwise keeping the two PRs in sync between two different people will probably be... hard.

>  My experience with 3rd Party middleware is that people either provide no trace info or bring whatever flavor of logger they see fit and that can be troublesome in terms of integration.

That's a tricky thing this PR does not solve. For example, this will not allow any of the expressjs org middleware to use it, because they are all made to be completely self-contained, no work on anything that provides a Node.js core `req`/`res` object without depending on things attached from in Express itself. If that is a goal, we'll probably need to discuss the best way to accomplish this at the next Express TC meeting and circle back around here to provide some feedback towards that goal.